### PR TITLE
taskt.Core.Server Cleanup

### PR DESCRIPTION
**Pull request by melugi**
* Models are now split from the HttpServerClient and live in their own Models folder.
* InvalidSocketUriException split from the LocalTcpListener.
* LocalTcpListener renamed to LocalTcpClient to better reflect that it does more than just listen.

Apart from these there were also many changes which removed redundant logic and checking within the clients themselves. And of course there's the usual cleanup like variable naming, visibility modifications, etc.

### CODE REVIEW

File: taskt/Core/Server/Models/BotStoreRequest.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+
+namespace taskt.Core.Server.Models
+{
+    public class BotStoreRequest
+    {
+        public Guid workerID { get; set; }
```
Comment (openbots-ff) [inline taskt/Core/Server/Models/BotStoreRequest.cs]: Please make these PascalCase

File: taskt/Core/Server/WebSocketClient.cs [hunk 2]
```diff
@@ -17,63 +20,61 @@ namespace taskt.Core.Server
...
+        public static frmScriptBuilder AssociatedBuilder;
+        public static string ConnectionException;
+        public static Logger SocketLogger;
+
+        private static string _publicKey;
+        private static string _serverURI;
+        private static DateTime _connectionOpened;
+        private static WebSocket _webSocket;
+        private static bool _retryOnFail;
+        private static System.Timers.Timer _reconnectTimer;
```
Comment (openbots-ff) [inline taskt/Core/Server/WebSocketClient.cs:32]: Please remove namespace

File: taskt/Core/Server/HttpServerClient.cs [hunk 3]
```diff
@@ -16,14 +17,14 @@ namespace taskt.Core.Server
     /// </summary>
     public static class HttpServerClient
     {
-        public static UI.Forms.frmScriptBuilder associatedBuilder;
-        private static Serilog.Core.Logger httpLogger;
-        private static System.Timers.Timer heartbeatTimer { get; set; }
-        private static ApplicationSettings appSettings { get; set; }
+        public static frmScriptBuilder associatedBuilder;
```
Comment (openbots-ff) [inline taskt/Core/Server/HttpServerClient.cs]: Please make PascalCase

File: taskt/Core/Server/HttpServerClient.cs [hunk 4]
```diff
@@ -86,40 +86,41 @@ public static bool GetGuid(bool overrideExisting = false)
...
+                    var deserialized = JsonConvert.DeserializeObject<Worker>(content);
+
+                    _appSettings.ServerSettings.HTTPGuid = deserialized.WorkerID;
+                    new ApplicationSettings().Save(_appSettings);
                     return true;
                 }
                 else
                 {
-                    httpLogger.Information("Client has an assigned GUID or override existing GUID set to false. To retrieve a new GUID, the existing setting must be cleared.");
+                    _httpLogger.Information("Client has an assigned GUID or override existing GUID set to false. To retrieve a new GUID, the existing setting must be cleared.");
```
Comment (openbots-ff) [inline taskt/Core/Server/HttpServerClient.cs]: Too long maybe?

**Review by openbots-ff — CHANGES_REQUESTED** [review]

File: taskt/Core/Server/WebSocketClient.cs (hunk 2 above)
Comment (melugi) [reply taskt/Core/Server/WebSocketClient.cs:32]: There's a conflict between `Windows.Forms.Timer` and `System.Timers.Timer` so I decided to just leave them. The other option would be to alias the namespaces which could be worth it if there was some meaningful name apart from `SystemTimer` or `FormsTimer`, but I think having the full assembly carries the meaning just as well. What do you think?

File: taskt/Core/Server/HttpServerClient.cs (hunk 3 above)
Comment (melugi) [reply taskt/Core/Server/HttpServerClient.cs]: Done.

File: taskt/Core/Server/Models/BotStoreRequest.cs (hunk 1 above)
Comment (melugi) [reply taskt/Core/Server/Models/BotStoreRequest.cs]: Done, thanks for catching them.

File: taskt/Core/Server/HttpServerClient.cs (hunk 4 above)
Comment (melugi) [reply taskt/Core/Server/HttpServerClient.cs]: Agree'd, made the change and split it up.

File: taskt/Core/Server/WebSocketClient.cs (hunk 2 above)
Comment (openbots-ff) [reply taskt/Core/Server/WebSocketClient.cs:32]: I vote to leave it as it is if there's an overlap. These things happen often enough.

**Review by openbots-ff — APPROVED** [review]

**Review by CameronHerwig — APPROVED** [review]

**Review by CameronHerwig — APPROVED** [review]

**Pull request merged**